Source/WebCore/platform/graphics/texmap/TextureMapperFPSCounter.cpp

```diff
@@ -61,10 +64,13 @@ void TextureMapperFPSCounter::updateFPSAndDisplay(TextureMapper& textureMapper,
     m_frameCount++;
     Seconds delta = MonotonicTime::now() - m_fpsTimestamp;
     if (delta >= m_fpsInterval) {
+        if (m_isShowingFPS)
+            m_currentFPS = m_frameCount / delta.seconds();
         m_lastFPS = int(m_frameCount / delta.seconds());
```

Comment:
It's a bit confusing that last and current are always the same value. I still think that TextureMapperFPSCounter has been designed to draw the fps on the compositing surface. It would be clearer for me to move the loging and sysprof counter directly to ThreadedCompositor and not use TextureMapperFPSCounter. If there's any port using threaded compositor that still wants to draw the fps we can keep the current fps counter there with an ifdef